### PR TITLE
Batch add new

### PR DESCRIPTION
**Pull request by tpfau**
Added batch addition of Reactions and metabolites.
addMetabolites now calls the batch addition, i.e. only one file to maintain.
Batch addition of reactions will NOT check the reactions for duplication (except for IDs). 

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: src/reconstruction/refinement/addMultipleReactions.m [hunk 1]
```diff
@@ -0,0 +1,169 @@
+function newmodel = addMultipleReactions(model,rxnIDs,metList,Stoichiometries,varargin)
+%Add a batch of reactions to the model.
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleReactions.m]: May you give here a more explicit explanation? Also, maybe refer to `addReaction`.

File: src/reconstruction/refinement/addMultipleReactions.m [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+%
+%    newmodel = addMultipleReactions(model,rxnIDs,metList,Stoichiometries,varargin)
+%
+% INPUTS:
+%    model:             The model to add the Metabolite batch to.
+%    rxnIDs:            The IDs of the reactions that shall be added.
+%    metList:           A list of metabolites for which stoichiometric coefficients are given
+%    Stoichiometries:   A Stoichiometric matrix of dimension
+%                       numel(metList) x numel(rxnID).
+%    varargin:          fieldName, Value pairs. The given fields will be set
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleReactions.m]: This should be `OPTIONAL INPUT`. Also, may you give a list of parameters?

File: src/reconstruction/refinement/addMetabolite.m [hunk 3]
```diff
@@ -92,129 +92,26 @@
...
+translation = {'b','b',false;...
+               'csense','csense',false;...
+               'metName','metNames',true;...
+               'metFormula','metFormulas',true;...
+               'ChEBIID','metChEBIID',true;...
+               'KEGGId','metKEGGID',true;...
+               'PubChemID','metPubChemID',true;...
+               'InChi','metInChIString',true;...
+               'Charge','metCharges',false};
+multiArgs = {};
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMetabolite.m]: May you add here a few lines explaining what this code does?

File: src/reconstruction/refinement/addMultipleMetabolites.m [hunk 4]
```diff
@@ -0,0 +1,79 @@
+function newmodel = addMultipleMetabolites(model,metIDs,varargin)
+%Add a batch of metabolites to the model.
+% USAGE:
+%
+%    model = addMultipleMetabolites(model,metIDs,varargin)
+%
+% INPUTS:
+%    model:     The model to add the Metabolite batch to.
+%    metIDs:    The metabolite IDs to add
+%    varargin:  fieldName, Value pairs. the given fields will be set
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleMetabolites.m]: This should be `OPTIONAL INPUT`. Also, may you give a list of parameters?

File: src/reconstruction/refinement/addMultipleMetabolites.m [hunk 5]
```diff
@@ -0,0 +1,79 @@
+function newmodel = addMultipleMetabolites(model,metIDs,varargin)
+%Add a batch of metabolites to the model.
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleMetabolites.m]: May you give here a more explicit explanation? Also, maybe refer to `addMetabolite`.

File: src/reconstruction/refinement/addMultipleMetabolites.m [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+if checkDuplicate && (any(ismember(model.mets,metIDs)) || numel(unique(metIDs)) < numel(metIDs))
+    error('Duplicate Metabolite ID detected.');
+end
+
+nMets = numel(model.mets);
+
+%We have make sure, that the new fields are in sync, so we create those
+%first.
+fieldDefs = getDefinedFieldProperties();
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleMetabolites.m]: May you explain here what this code does?

File: src/reconstruction/refinement/addMultipleMetabolites.m [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+fieldDefs = fieldDefs(cellfun(@(x) strcmp(x,'mets'), fieldDefs(:,2)) | cellfun(@(x) strcmp(x,'mets'), fieldDefs(:,3)));
+modelMetFields = getModelFieldsForType(model,'mets');
+
+%First, check which fields already exist, and create those which don't
+%exist yet.
+
+model.mets = [model.mets;columnVector(metIDs)];
+
+for field = 1:2:numel(varargin)
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleMetabolites.m]: May you explain here what this code does?

File: src/reconstruction/refinement/addMultipleReactions.m [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+if numel(unique(metList)) < numel(metList)
+    error('Duplicate Metabolite ID detected.');
+end
+
+nRxns = numel(model.rxns);
+
+%We have make sure, that the new fields are in sync, so we create those
+%first.
+fieldDefs = getDefinedFieldProperties();
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleReactions.m]: May you explain here what this code does?

File: src/reconstruction/refinement/addMultipleMetabolites.m [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+%
+%    model = addMultipleMetabolites(model,metIDs,varargin)
+%
+% INPUTS:
+%    model:     The model to add the Metabolite batch to.
+%    metIDs:    The metabolite IDs to add. No duplicate IDs may be provided
+%               and no ID may be present in the supplied model.
+%
+% OPTIONAL INPUTS:
+%    varargin:  fieldName, Value pairs.
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addMultipleMetabolites.m]: What are the possible options here?

Comment (tpfau) [reply src/reconstruction/refinement/addMultipleMetabolites.m]: all valid field names associated with mets specified in the Field Spec.

**Comment by laurentheirendt** [comment]
Thanks @tpfau ⚽️

**Pull request merged**